framework/src/main/java/org/checkerframework/framework/util/typeinference8/util/Java8InferenceContext.java

```diff
@@ -119,6 +122,15 @@ public int getNextCaptureVariableId() {
     return captureVariableCount++;
   }
 
+  /**
+   * Returns the next number to use as the id for a qualifier variable.
+   *
+   * @return the next number to use as the id for a qualifier variable
+   */
+  public int getNextQualifierVariableId() {
```

Comment:
I'm curious why this method is here rather than in the class(es) that represent qualifier variables.

Reply:
These ids are supposed to be unique for the current inference problem. One `Java8InferenceContext` is created for each inference problem, so that's why it's in this class.  (I've tried to clarify the Javadoc.)